src/account/accountEditor.component.js

```diff
@@ -10,74 +10,65 @@ import RaisedButton from 'material-ui/RaisedButton';
 import appActions from '../app.actions';
 import accountActions from './account.actions';
 
-function isNotEmpty(value) {
-    return value && String(value).trim().length > 0;
-}
-
 
-class AccountEditor extends React.Component {
-    constructor(props, context) {
-        super(props, context);
-
-        this.state = Object.assign({}, {
+class AccountEditor extends Component {
+    constructor (props) {
+        super(props);
+        this.state = {
             oldPassword: '',
             newPassword: '',
             reNewPassword: '',
             wrongOldPasswordText: '',
-        });
-
-        this.d2 = this.context.d2;
-        this.getTranslation = this.d2.i18n.getTranslation.bind(this.d2.i18n);
-        this.updatePassword = this.updatePassword.bind(this);
-        this.updateState = this.updateState.bind(this);
-        this.isSamePassword = this.isSamePassword.bind(this);
-        this.isVerifiedPassword = this.isVerifiedPassword.bind(this);
-        this.isVerifiedPassword.message = 'wrong_old_password';
-        this.clearRepeatPassword = this.clearRepeatPassword.bind(this);
+        };
+        this.isVerifiedPassword.message = "wrong_old_password";
     }
-
-    isSamePassword(value) {
+    
+    isSamePassword = (value) => {
         return value === this.state.newPassword;
     }
 
-    isVerifiedPassword(value) {
-        const api = this.d2.Api.getApi();
+    isNotEmpty = (value) => {
+        return value && String(value).trim().length > 0;
+    }
+
+    isVerifiedPassword = (value) => {
+        const api = this.context.d2.Api.getApi();
         return new Promise((resolve, reject) => {
-            api.post('me/verifyPassword', { password: value })
+            api.post('/29/me/verifyPassword', { password: value })
```

Comment:
I see that you are hardcoding the API version in each endpoint, this is gonna be annoying to update the next time.

Take a look at how it's solved here: https://github.com/dhis2/dashboards-app/blob/master/package.json#L95-L97

Maybe you can use the same trick?